src/hooks/useCourtContracts.js

```diff
@@ -351,6 +351,32 @@ export function useAppealFeeAllowance(owner) {
   return allowance
 }
 
+export function useActiveBalanceOfAt(juror, termId) {
+  const jurorRegistryContract = useCourtContract(
+    CourtModuleType.JurorsRegistry,
+    jurorRegistryAbi
+  )
+  const [activeBalance, setActiveBalance] = useState(bigNum(-1))
+
+  useEffect(() => {
+    const getActiveBalanceOfAt = async () => {
+      if (!jurorRegistryContract) return
+
+      retryMax(() => jurorRegistryContract.activeBalanceOfAt(juror, termId))
+        .then(balance => {
+          setActiveBalance(balance)
+        })
+        .catch(err => {
+          console.error(`Error fetching active balance for juror : ${err}`)
+        })
+    }
+
+    getActiveBalanceOfAt()
```

Comment:
We should use the ["cancel" pattern](https://github.com/aragon/court-dashboard/blob/development/src/hooks/useTokenBalanceToUsd.js#L89) here, where we avoid setting state if the hook has been re-run.

If we're using timers to retry, we should also clear them in the return of the effect.

Reply:
Yeah good point, will add the pattern to all these type of functions.